internal/util/kubernetes/kubernetes.go

```diff
@@ -237,7 +237,7 @@ func GetResource(
 // ApplyCRSpec either creates or updates an object identified by the RawExtension, using the new definition,
 // first checking to see if there's a difference in Spec before applying
 // TODO: use CreateCR and UpdateCR instead
-func ApplyCRSpec(ctx context.Context, restConfig *rest.Config, object *GenericObject, pluralName string, specHash string, rolloutHashAnnotation string) error {
+func ApplyCRSpec(ctx context.Context, restConfig *rest.Config, object *GenericObject, pluralName string, currentRolloutSpecHash string, previousRolloutSpecHash string, childSpecHash string) error {
```

Comment:
We don't have to do it in this PR (I'd be happy to refactor later), but I'd like to see this function go back to being more of a pure function that's not aware of hashes/rollouts and simply "applies CRs"

Reply:
refactored